src/main/java/org/opensearch/geospatial/ip2geo/processor/Ip2GeoCache.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geospatial.ip2geo.processor;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+
+import org.opensearch.common.cache.Cache;
+import org.opensearch.common.cache.CacheBuilder;
+import org.opensearch.common.unit.TimeValue;
+
+/**
+ * The in-memory cache for the ip2geo data. There should only be 1 instance of this class.
+ */
```

Comment:
How we are enforcing a single instance creation of this class?

Reply:
We are creating the cache in createComponents method of plugin and pass it to Ip2Geo factory.